cp-attendance-hybrid/src/modules/home/home.page.tsx

```diff
@@ -30,20 +30,33 @@ export const Home: React.FC<HomeScreenProps> = ({ navigation }) => {
   const [errorLesson, setErrorLesson] = useState<Error>();
   const [attendanceModalOpen, setAttendanceModalOpen] = useState(false);
   const [lessonId, setLessonId] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [refreshing, setRefreshing] = useState(false);
   const user = useUser();
 
-  useEffect(() => {
-    const getLessons = async () => {
-      const lessonVector = await LessonService.listLessons(String(user.user?.studentClass));
+  const getLessons = useCallback(async () => {
+    setLoading(true);
+    const lessonVector = await LessonService.listLessons(String(user.user?.studentClass));
       if (!(lessonVector instanceof Error)){
         const filteredArray = lessonVector.filter(lesson => isToday(lesson.startDatetime) && lesson.endDatetime > new Date())
         setLessons(filteredArray);
       } 
       else setErrorLesson(lessonVector);
-    }
+      setLoading(false);
+  }, []);
```

Comment:
Aqui acho que precisa incluir user como dependência do useCallback, porque se ele mudar por algum caralho de motivo n sei se a função re-renderizaria

```suggestion
  }, [user]);
```